README.md

```diff
@@ -32,7 +32,7 @@ These differences are:
 
 In order to build Wire for Android locally, it is necessary to have the following tools installed:
 
-- JDK 17
+- JDK 21
```

Comment:
This doesn't actually make a difference.

Gradle itself is gonna run on your JDK of preference. It should run on JDK 17 or 21 fine.
However we are facing some issues with verify DB migrations on JDK 21.

With the change to use `jvmToolchain` settings, Gradle will download the necessary JDK when compiling/running Java tasks.